extensions/amp-story/0.1/bookend-share.js

```diff
@@ -0,0 +1,179 @@
+    class: 'i-amphtml-story-share-name',
+  });
+
+  // constant value, no XSS risk
+  iconEl./*OK*/innerHTML = ICONS.link;
+
+  // TODO(alanorozco): i18n
+  nameEl.textContent = 'Get Link';
+
+  root.appendChild(iconEl);
+  root.appendChild(nameEl);
+
+  fragment.appendChild(root);
+
+  return fragment;
+}
+
+
+function buildProvider(doc, type, opt_params) {
+  const fragment = doc.createDocumentFragment();
+  const root = doc.createElement('li');
+
+  const shareEl = createElementWithAttributes(doc, 'amp-social-share', {
+    type,
+    width: 48,
+    height: 48,
+    class: 'i-amphtml-story-share-icon',
+  });
+
+  if (opt_params) {
+    Object.keys(opt_params).forEach(field =>
+        shareEl.setAttribute(`data-param-${field}`, opt_params[field]));
+  }
+
+  if (type == 'email') {
+    // constant value, no XSS risk
+    shareEl./*OK*/innerHTML = ICONS.mail;
+  }
+
+  const nameEl = createElementWithAttributes(doc, 'span', {
+    class: 'i-amphtml-story-share-name',
+  });
+
+  nameEl.textContent = SHARE_PROVIDER_NAME[type] || type;
+
+  root.appendChild(shareEl);
+  root.appendChild(nameEl);
+
+  fragment.appendChild(root);
+
+  return fragment;
+}
+
+
+export class BookendShareWidget {
+  /** @param {!Window} win */
+  constructor(win) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {?Element} */
+    this.root_ = null;
+  }
+
+  /** @param {!Window} win */
+  static create(win) {
+    return new BookendShareWidget(win);
+  }
+
+  /** @return {!Element} */
+  build() {
+    dev().assert(!this.root_, 'Already built.');
+
+    this.root_ = createElementWithAttributes(this.win_.document, 'ul', {
+      'class': 'i-amphtml-story-share-list',
+    });
+
+    this.add_(buildLinkShareItem(this.win_.document));
+
+    this.maybeAddNativeShare_();
+
+    return this.root_;
+  }
+
+  /** @private */
+  maybeAddNativeShare_() {
+    // TODO(alanorozco): Implement
+  }
+
+  /**
+   * @param {!Array<!JsonObject>} providers
+   * @public
+   */
+  // TODO(alanorozco): Set story metadata in share config
+  setProviders(providers) {
+    const fragment = this.win_.document.createDocumentFragment();
+
+    this.loadRequiredExtensions_();
+
+    Object.keys(providers).forEach(type => {
+      if (isObject(providers[type])) {
+        fragment.appendChild(buildProvider(this.win_.document, type,
+            /** @type {!JsonObject} */ (providers[type])));
+        return;
+      }
+
+      // Bookend config API requires real boolean, not just truthy
+      if (providers[type] === true) {
+        fragment.appendChild(buildProvider(this.win_.document, type));
+        return;
+      }
+
+      user().warn('AMP-STORY',
+          'Invalid amp-story bookend share configuration for %s. ' +
+          'Value must be `true` or a params object.',
+          type);
+    });
+
+    this.add_(fragment);
+  }
+
+  /** @private */
+  loadRequiredExtensions_() {
+    Services.extensionsFor(this.win_).loadExtension('amp-social-share');
```

Comment:
Done.

All STAMPs are required to have a bookend, but not all bookends will have share buttons that require `amp-social-share`, since that is configurable.

It's also not guaranteed that the user will navigate to the bookend, so the extension is only loaded once `Bookend` sets share configuration for the share widget. This happens on "bookend preload time", which is executed once the user gets close enough to the widget. 

This will be clear in the upcoming PR for the top-level `AmpStory` component.